File: hackerrank/30-days-of-code/recursion/recursion.py
#!/usr/bin/env python

import math
import os
import random
import re
import sys

# Complete the factorial function below.
def factorial(n):
    value = 1
    for i in range(2, n + 1):
        value = i * factorial(n-1)
    return value

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input())

    result = factorial(n)

    fptr.write(str(result) + '\n')

    fptr.close()

File: hackerrank/30-days-of-code/recursion/recursion.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the factorial function below.
int factorial(int n) {
    int value = 1;
    for (int i = 2; i < n+1; i++)
    {
        value = i * factorial(n-1);
    }

    return value;
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    int n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    int result = factorial(n);

    fout << result << "\n";

    fout.close();

    return 0;
}
